### PR TITLE
[SPARK-29108][SQL][TESTS] Port window.sql (Part 2)

### PR DESCRIPTION
**Pull request by DylanGuedes**
### What changes were proposed in this pull request?

This PR ports window.sql from PostgreSQL regression tests https://github.com/postgres/postgres/blob/REL_12_STABLE/src/test/regress/sql/window.sql from lines 320~562

The expected results can be found in the link: https://github.com/postgres/postgres/blob/REL_12_STABLE/src/test/regress/expected/window.out

## How was this patch tested?
Pass the Jenkins.

### Why are the changes needed?
To ensure compatibility with PGSQL

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Comparison with PgSQL results.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
add to whitelist

**Comment by HyukjinKwon** [comment]
cc @maropu, @dongjoon-hyun, @wangyum, @MaxGekk

**Comment by gengliangwang** [comment]
@DylanGuedes Thanks for the work.
Please wait until #26107 is merged.

**Comment by dongjoon-hyun** [comment]
#26107 is merged. Thanks, @gengliangwang .

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Review by HyukjinKwon — APPROVED** [review]
Seems fine. I'm going to merge this in few days.

File: sql/core/src/test/resources/sql-tests/inputs/postgreSQL/window_part2.sql [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+-- select sum(salary) over (order by enroll_date range between '1 year' preceding and '1 year' following
+--   exclude ties), salary, enroll_date from empsalary;
+
+-- [SPARK-27951] ANSI SQL: NTH_VALUE function
+-- select first_value(salary) over(order by salary range between 1000 preceding and 1000 following),
+-- lead(salary) over(order by salary range between 1000 preceding and 1000 following),
+-- nth_value(salary, 1) over(order by salary range between 1000 preceding and 1000 following),
+-- salary from empsalary;
+
+select last(salary) over(order by salary range between 1000 preceding and 1000 following),
```
[the remark this answers is not in the document]
Comment (DylanGuedes) [reply sql/core/src/test/resources/sql-tests/inputs/postgreSQL/window_part2.sql]: I can make them uppercase, but, are you sure? I think that in PostgreSQL they are not always uppercase, such that the diff will be totally different.

**Comment by HyukjinKwon** [comment]
I'm going to just merge this. It has been open so long and I am sure it has been exposed to reviewers enough time.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**